src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs

```diff
@@ -969,7 +969,7 @@ public async Task SslStream_RandomSizeWrites_OK(int bufferSize, int readBufferSi
                         int writeLength = Math.Min(data.Length, writeBufferSize);
                         if (useAsync)
                         {
-                            server.WriteAsync(data.Slice(0, writeLength)).GetAwaiter().GetResult();
+                            server.WriteAsync(data.Slice(0, writeLength)).AsTask().GetAwaiter().GetResult();
```

Comment:
Can we just make the lambda be `Task.Run(async () =>` and then await this rather than synchronously block?  Same for the method below.

Reply:
We could and I had code with `await` when I was experimenting. But I'm wondering if we _should_ have some test coverage for suboptimal cases too e.g. keep the `GetAwaiter` here on purpose.

Reply:
> But I'm wondering if we should have some test coverage for suboptimal cases too e.g. keep the GetAwaiter here on purpose.

We could, though I think that should be a dedicated test if we want that.